feat: add `SpectateCommunity` api

osmaczko opened the pull request:
- added `SpectateCommunity` endpoint, it is supposed to be used in scenarios where we want to "Go to public Community" and see its content without joining
- added `spectated` field to `Community`, it means we are observing the community and its chats but we are not members

Use case:
https://github.com/status-im/status-desktop/issues/7072#issuecomment-1246560885

status-desktop: https://github.com/status-im/status-desktop/pull/7514

osmaczko committed:
feat: add `SpectateCommunity` api

- added `SpectateCommunity` endpoint, it is supposed to be used in
  scenarios where we want to "Go to public Community" and see its
  content without joining
- added `spectated` field to `Community`, it means we are observing the
  community and its chats but we are not members

Use case:
https://github.com/status-im/status-desktop/issues/7072#issuecomment-1246560885

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [x] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (6)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 834c646d | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2877/1/) | 2022-09-22 16:01:54 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220922-155916-834c64-pr2877.zip) |
| :heavy_check_mark: | 834c646d | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2877/1/) | 2022-09-22 16:03:17 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220922-155916-834c64-pr2877.zip) |
| :heavy_check_mark: | 834c646d | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2877/1/) | 2022-09-22 16:04:07 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220922-155911-834c64-pr2877.aar) |
| | | | | | | |
| :heavy_check_mark: | 18a07905 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2877/2/) | 2022-09-23 09:23:59 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220923-092106-18a079-pr2877.zip) |
| :heavy_check_mark: | 18a07905 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2877/2/) | 2022-09-23 09:24:29 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220923-092106-18a079-pr2877.aar) |
| :heavy_check_mark: | 18a07905 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2877/2/) | 2022-09-23 09:26:25 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220923-092106-18a079-pr2877.zip) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 480aee71 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2877/3/) | 2022-09-23 10:00:16 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220923-095808-480aee-pr2877.zip) |
| :heavy_check_mark: | 480aee71 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2877/3/) | 2022-09-23 10:03:16 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220923-095808-480aee-pr2877.aar) |
| :heavy_check_mark: | 480aee71 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2877/3/) | 2022-09-23 10:03:27 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220923-095808-480aee-pr2877.zip) |
| | | | | | | |
| :heavy_check_mark: | e9940fb6 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2877/4/) | 2022-10-06 18:12:12 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-221006-180947-e9940f-pr2877.zip) |
| :heavy_check_mark: | e9940fb6 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2877/4/) | 2022-10-06 18:13:47 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-221006-180947-e9940f-pr2877.zip) |
| :heavy_check_mark: | e9940fb6 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2877/4/) | 2022-10-06 18:14:54 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-221006-180942-e9940f-pr2877.aar) |

richard-ramos reviewed (approved):
The only changes I'd do is to not commit the changes on bindata.go / migrations.go files that werent affected by this commit, and maybe rename Spectate to Watch, but this comment can be ignored :)

osmaczko commented:
> The only changes I'd do is to not commit the changes on bindata.go / migrations.go files that werent affected by this commit, and maybe rename Spectate to Watch, but this comment can be ignored :)

:heavy_check_mark: removed changes to bindata.go / migrations.go not affected by this commit
:negative_squared_cross_mark: renamed spectate to watch, I like "spectator" more than "watcher" (maybe because I played [CS 1.6](https://counterstrike.fandom.com/wiki/Spectator) a lot :D)
